src/Dto/Segments/ListSegments.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OneSignal\Dto\Segments;
+
+use OneSignal\Dto\AbstractDto;
+
+class ListSegments implements AbstractDto
+{
+    protected ?int $limit = null;
+
+    protected ?int $offset = null;
+
+    public function __construct(?int $limit = null, ?int $offset = null)
+    {
+        $this->limit = $limit;
+        $this->offset = $offset;
+    }
+
+    public function setLimit(?int $limit = null): self
+    {
+        $this->limit = $limit;
+
+        return $this;
+    }
+
+    public function setOffset(?int $offset = null): self
```

Comment:
```suggestion
    /**
     * @param int<0, 2147483648>|null $offset
     */
    public function offset(?int $offset = null): self
```

same question applies